File: Simple connection methods for MS-Access using OleDb data provider/[C#][VB]-Simple connection methods for MS-Access using OleDb data provider/C# and VB/AccessConnections_vb/ConnectionsAccess.vb
Imports System.Data.OleDb

Public Module ConnectionsAccess
    Private providers As New Dictionary(Of String, String)() From {{".accdb", "Microsoft.ACE.OLEDB.12.0"}, {".mdb", "Microsoft.Jet.OLEDB.4.0"}}

    <System.Diagnostics.DebuggerStepThrough()>
    <System.Runtime.CompilerServices.Extension()>
    Public Function BuildConnectionString(ByVal DatabaseName As String) As String

        Dim Builder As OleDbConnectionStringBuilder = New OleDbConnectionStringBuilder With
            {
                .Provider = providers(IO.Path.GetExtension(DatabaseName).ToLower()),
                .DataSource = DatabaseName
            }

        Return Builder.ConnectionString

    End Function
End Module

File: Simple connection methods for MS-Access using OleDb data provider/[C#][VB]-Simple connection methods for MS-Access using OleDb data provider/C# and VB/AccessConnections_cs/ConnectionsAccess.cs
using System.IO;
using System.Data.OleDb;
using System.Collections.Generic;

namespace AccessConnections_cs
{
    public static class ConnectionsAccess
    {
        private static Dictionary<string, string> providers = new Dictionary<string, string>()
        {
            { ".accdb" , "Microsoft.ACE.OLEDB.12.0" },
            { ".mdb" , "Microsoft.Jet.OLEDB.4.0" }
        };

        public static string BuildConnectionString(this string DatabaseName)
        {

            OleDbConnectionStringBuilder Builder = new OleDbConnectionStringBuilder
            {
                Provider = providers[Path.GetExtension(DatabaseName).ToLower()],
                DataSource = DatabaseName
            };

            return Builder.ConnectionString;

        }
    }
}
